File: db/fsharp/SalesManagement.Domain/Models/Order.fs
namespace SalesManagement.Domain.Models

open System

/// <summary>
/// 受注データ
/// </summary>
[<CLIMutable>]
type Order = {
    OrderNo: string              // 受注番号（PK）
    OrderDate: DateTime          // 受注日
    CustomerCode: string         // 得意先コード（FK）
    CustomerBranch: int          // 得意先枝番（FK）
    EmployeeCode: string         // 社員コード（FK）
    DueDate: DateTime option     // 納期
    OrderAmount: int             // 受注金額合計
    ConsumptionTax: int          // 消費税合計
    SlipComment: string option   // 伝票備考
    DepartmentCode: string       // 部門コード（FK）
    CreatedAt: DateTime          // 作成日時
    CreatedBy: string            // 作成者名
    UpdatedAt: DateTime          // 更新日時
    UpdatedBy: string            // 更新者名
}

File: db/csharp/SalesManagement.Domain/Models/Order.cs
namespace SalesManagement.Domain.Models
{
    /// <summary>
    /// 受注データ
    /// </summary>
    public class Order
    {
        public string OrderNo { get; set; } = string.Empty;
        public DateTime OrderDate { get; set; }
        public string DepartmentCode { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public string CustomerCode { get; set; } = string.Empty;
        public int? CustomerBranch { get; set; }
        public string EmployeeCode { get; set; } = string.Empty;
        public DateTime? RequiredDate { get; set; }
        public string? CustomerOrderNo { get; set; }
        public string WarehouseCode { get; set; } = string.Empty;
        public int OrderAmount { get; set; }
        public int ConsumptionTax { get; set; }
        public string? SlipComment { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTime UpdatedAt { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
    }
}
